### PR TITLE
Fix file upload notifications in long-haul tests of service app

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

File: e2e/LongHaul/service/IotHub.cs [hunk 1]
```diff
@@ -165,32 +165,41 @@ AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
 
         public async Task ReceiveFileUploadAsync(CancellationToken ct)
         {
-            while (!ct.IsCancellationRequested)
```
Comment (brycewang-microsoft) [inline e2e/LongHaul/service/IotHub.cs:168]: We are registering file upload notification callback here, so we shouldn't do this in a loop.

File: e2e/LongHaul/service/IotHub.cs [hunk 2]
```diff
@@ -165,32 +165,41 @@ AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
...
+
+                s_blobContainerClient.DeleteBlobIfExists(fileUploadNotification.BlobName);
+                return Task.FromResult(ackType);
+            }
+
+            s_serviceClient.FileUploadNotifications.FileUploadNotificationProcessor = FileUploadNotificationCallback;
+
+            try
             {
                 await s_serviceClient.FileUploadNotifications.OpenAsync(ct).ConfigureAwait(false);
```
Comment (brycewang-microsoft) [inline e2e/LongHaul/service/IotHub.cs:192]: Callback should be registered before opening the connection.

File: e2e/LongHaul/service/IotHub.cs [hunk 3]
```diff
@@ -165,32 +165,41 @@ AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
...
-                s_serviceClient.FileUploadNotifications.FileUploadNotificationProcessor = FileUploadNotificationCallback;
-                _logger.Metric("TotalFileUploadNotificiationsReceived", _totalFileUploadNotificationsReceived);
-
-                await Task.Delay(s_receiveFileUploadInterval, ct).ConfigureAwait(false);
+                catch (OperationCanceledException) { }
+            }
+            finally
+            {
+                await s_serviceClient.FileUploadNotifications.CloseAsync().ConfigureAwait(false);
             }
```
Comment (brycewang-microsoft) [inline e2e/LongHaul/service/IotHub.cs:203]: Similar to what we are doing in [`ReceiveMessageFeedbacksAsync()`](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/e2e/LongHaul/service/IotHub.cs#L154-L163).

File: e2e/LongHaul/service/IotHub.cs (hunk 1 above)
Comment (drwill-ms) [reply e2e/LongHaul/service/IotHub.cs:168]: Correct, but we need to have something wait on the cancellation token before the method exits.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**